File: educational-core/src/com/jetbrains/edu/learning/stepik/courseFormat/StepikSection.java
package com.jetbrains.edu.learning.stepik.courseFormat;

import com.jetbrains.edu.learning.courseFormat.Section;
import com.jetbrains.edu.learning.courseFormat.remote.RemoteInfo;
import com.jetbrains.edu.learning.stepik.courseFormat.remoteInfo.StepikSectionRemoteInfo;
import org.jetbrains.annotations.NotNull;

public class StepikSection extends Section {

  public StepikSection() {
    setRemoteInfo(new StepikSectionRemoteInfo());
  }

  @NotNull
  public StepikSectionRemoteInfo getStepikRemoteInfo() {
    final RemoteInfo info = super.getRemoteInfo();
    assert info instanceof StepikSectionRemoteInfo;
    return (StepikSectionRemoteInfo)info;
  }
}

File: educational-core/src/com/jetbrains/edu/learning/stepik/courseFormat/StepikSection.kt
package com.jetbrains.edu.learning.stepik.courseFormat

import com.jetbrains.edu.learning.courseFormat.Section
import com.jetbrains.edu.learning.stepik.courseFormat.remoteInfo.StepikSectionRemoteInfo

class StepikSection : Section() {

  val stepikRemoteInfo: StepikSectionRemoteInfo
    get() {
      val info = super.getRemoteInfo()
      assert(info is StepikSectionRemoteInfo)
      return info as StepikSectionRemoteInfo
    }

  init {
    remoteInfo = StepikSectionRemoteInfo()
  }
}
